### PR TITLE
feature #3616 - "Set your account" password link in SMS activation messages

### LINKED ISSUE #3616: Send a "Set your password" link in account activation SMS message
Dependent on #1784 and #1782

**Description**
After #1784, new volunteers/supervisors/admins should receive an SMS after their account is created with the following message:
_A CASA [volunteer/supervisor/admin] account was created for you. Visit [link] to change your text message settings._

For this issue, we want to append _Set your password: [link]_ to the SMS. So, the new SMS message should be like:
_A CASA [volunteer/supervisor/admin] account was created for you. First, set your password here [link]. Then visit [link] to change your text message settings._

Both links should be shortened.

_QA Login Details:_  
[Link to QA site](https://casa-qa.herokuapp.com/)  

Login Emails: 
- volunteer1@example.com  view site as a volunteer
- supervisor1@example.com view site as a supervisor
- casa_admin1@example.com view site as an admin

password for all users: 12345678  

### Questions? Join Slack!

We highly recommend that you join us in slack https://rubyforgood.herokuapp.com/ #casa channel to ask questions quickly and hear about office hours (currently Tuesday 6-8pm Pacific), stakeholder news, and upcoming new issues.

### PR DESCRIPTION
**Pull request by xihai01**
### What github issue is this PR for, if any?
Resolves #3616 
A set up password link is now sent with the account creation SMS. In addition, this link and the "/users/edit" link to change SMS settings are shortened using short.io service.

Note: Short io credentials do not exist in production env yet, so the SMS links aren't sent. In cases where such as these when there are issues with short io API itself, a default msg is sent instead as shown in screenshot

### What changed, and why?
- created a function called handle_short_urls() for calling short io service to return a hash of shortened links
- refactor SMS body message function to accept the hash and render returned messages dynamically based on the values inside the hash
- refactor volunteer/super/admin controller create actions to handle short links in SMS

### How is this tested? (please write tests!) 💖💪
- refactor controller request specs to assert for short io service calls
- add tests to application_controller specs for handle_short_urls()
- add tests to spec/helpers to verify SMS body message functions return correct msgs

### Screenshots please :)
SMS when short link service has issues
coming soon

### Feelings gif (optional)
coming soon
`alt text`

### Feedback please? (optional)
We are very interested in your feedback! Please give us some :) https://forms.gle/1D5ACNgTs2u9gSdh9

### CODE REVIEW

File: .allow_skipping_tests [hunk 1]
```diff
@@ -47,6 +47,8 @@ helpers/all_casa_admins/casa_orgs_helper.rb
 helpers/contact_types_helper.rb
 helpers/date_helper.rb
 helpers/template_helper.rb
+helpers/api_base_helper.rb
```
Comment (compwron) [inline .allow_skipping_tests:50]: Please write tests for code instead of ignoring them

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -30,17 +30,30 @@ def after_sign_out_path_for(resource_or_scope)
 
   protected
 
+  def handle_short_url(url_list)
+    hash_of_short_urls = {}
+    url_list.each_with_index { |val, index|
+      # call short io service to shorten url
```
Comment (compwron) [inline app/controllers/application_controller.rb:36]: good comment

**Review by compwron — APPROVED** [review]

**Comment by compwron** [comment]
safe to merge

File: .allow_skipping_tests (hunk 1 above)
Comment (xihai01) [reply .allow_skipping_tests:50]: got it

**Pull request merged**